### ISSUE TITLE
Unable to enter a newline if Shift is held

### ISSUE THREAD

**Issue by chlorocodes** [zed-industries/zed#5144]
### Check for existing issues

- [X] Completed

### Problem

I don't actually know if most would consider this a problem, but I am unable to insert a newline via <kbd>Enter</kbd> if <kbd>Shift</kbd> is held at the same time. I've been running into this issue a few times now while using Zed. There are often situations where you need to hold the <kbd>Shift</kbd> key for some characters and then put in a newline afterwards. Example:

```js
function() { // Hold Shift to type in "() {" and then hit Enter to insert a newline
}
```

I know technically one shouldn't be holding the <kbd>Shift</kbd> after they have finished typing `() {` but it's something that can happen pretty easily if typing fast and I don't think it should prevent newlines from being inserted. I tested the behavior in all the other editors I've used previously (Sublime/Atom/VSCode), and they didn't seem to have this problem (you can see the video below).

### Steps to reproduce

1. Hold the <kbd>Shift</kbd> key
1. Press <kbd>Enter</kbd> to insert a newline (doesn't work)

### Expected behavior

I was expecting <kbd>Enter</kbd> to insert a newline even if I am holding <kbd>Shift</kbd>

### Environment

Zed 0.62.7 – /Applications/Zed.app
macOS 12.0.1
architecture x86_64

I have tested with Vim mode on/off, both seem to have this issue

### If applicable, add mockups / screenshots to help explain present your vision of the feature

https://user-images.githubusercontent.com/112922394/200159100-5009b6aa-de41-4208-a55d-ccb8cc34253d.mov

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue

N/A

**Comment by maxdeviant** [comment]
@chlorocodes I can no longer reproduce this in Zed v0.119.19.

Could you see if this issue is resolved in the latest version?

**Comment by chlorocodes** [comment]
hi @maxdeviant sorry for getting back to this late. The issue looks fixed now, thanks a lot! (Tested in Zed 0.134.4).

Closing the issue
